docs/source/notes/windows.rst

```diff
@@ -0,0 +1,256 @@
+Windows Usage
+==========================
+
+Building from source
+--------------------
+
+Include optional components
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There're two supported components for Windows PyTorch:
+MKL and MAGMA. Here are the steps to build with them.
+
+.. code-block:: bat
+
+    REM Make sure you has 7z and curl installed.
+
+    REM Download MKL files
+    curl https://s3.amazonaws.com/ossci-windows/mkl_2018.2.185.7z -k -O
+    7z x -aoa mkl_2018.2.185.7z -omkl
+
+    REM Download MAGMA diles
+    REM cuda90/cuda91 is also available in the following line.
+    set CUDA_PREFIX=cuda80 
+    curl -k https://s3.amazonaws.com/ossci-windows/magma_%CUDA_PREFIX%_release_mkl_2018.2.185.7z -o magma.7z
+    7z x -aoa magma.7z -omagma
+    
+    REM Setting essential environment variables
+    set "CMAKE_INCLUDE_PATH=%cd%\\mkl\\include"
+    set "LIB=%cd%\\mkl\\lib;%LIB%"
+    set "MAGMA_HOME=%cd%\\magma"
+
+Speeding CUDA build for Windows
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Visual Studio doesn't support parallel custom task currently.
+As an alternative, we can use ``Ninja`` to parallelize CUDA
+build tasks. It can used by typing only a few lines of code.
+
+.. code-block:: bat
+    
+    REM Let's install ninja first.
+    pip install ninja
+
+    REM Set it as the cmake generator
+    set CMAKE_GENERATOR=Ninja
+
+
+One key install script
+^^^^^^^^^^^^^^^^^^^^^^
+
+You can take a look at the script `here
+<https://github.com/peterjc123/pytorch-scripts>`_. 
+It will lead the way for you.
```

Comment:
Is building from source really as compilcated as having a bunch of build scripts? What do they do, and why isn't is as simple as `python setup.py install`, like on Linux/macOS?